Source/Core/VideoCommon/RenderBase.cpp

```diff
@@ -589,21 +589,38 @@ void Renderer::CheckForConfigChanges()
 // Create On-Screen-Messages
 void Renderer::DrawDebugText()
 {
-  if (g_ActiveConfig.bShowFPS)
+  if (g_ActiveConfig.bShowFPS || g_ActiveConfig.bShowVPS || g_ActiveConfig.bShowSpeed)
   {
     // Position in the top-right corner of the screen.
     ImGui::SetNextWindowPos(ImVec2(ImGui::GetIO().DisplaySize.x - (10.0f * m_backbuffer_scale),
-                                   10.0f * m_backbuffer_scale),
+                                   10.f * m_backbuffer_scale),
                             ImGuiCond_Always, ImVec2(1.0f, 0.0f));
-    ImGui::SetNextWindowSize(ImVec2(100.0f * m_backbuffer_scale, 30.0f * m_backbuffer_scale));
 
-    if (ImGui::Begin("FPS", nullptr,
+    int count = g_ActiveConfig.bShowFPS + g_ActiveConfig.bShowVPS + g_ActiveConfig.bShowSpeed;
+    ImGui::SetNextWindowSize(
+        ImVec2(85.f * m_backbuffer_scale, (14.f + 16.f * count) * m_backbuffer_scale));
+
+    if (ImGui::Begin("Performance", nullptr,
                      ImGuiWindowFlags_NoTitleBar | ImGuiWindowFlags_NoInputs |
                          ImGuiWindowFlags_NoMove | ImGuiWindowFlags_NoSavedSettings |
                          ImGuiWindowFlags_NoScrollbar | ImGuiWindowFlags_NoNav |
                          ImGuiWindowFlags_AlwaysAutoResize | ImGuiWindowFlags_NoFocusOnAppearing))
     {
-      ImGui::TextColored(ImVec4(0.0f, 1.0f, 1.0f, 1.0f), "FPS: %.2f", m_fps_counter.GetFPS());
+      const double fps = m_fps_counter.GetFPS();
+      const double vps = m_vps_counter.GetFPS();
+      const double speed = 100.0 * vps / VideoInterface::GetTargetRefreshRate();
+
+      // Change Color based on % Speed
+      const float r = 1.0 - (speed - 80.0) / 20.0;
+      const float g = speed / 80.0;
+      const float b = (speed - 90.0) / 10.0;
```

Comment:
Since it seems my question seems to have drowned in the noise, I'll make it a line comment here:
> Just checking, whats the colors do when we exceed 100%?

At least by mathing it out, anything above `speed = 100.0` will yield a negative `r`; and I'm not sure if thats a good thing (as far as `ImVec4` and `ImGui::TextColored` are concerned).

Reply:
I have tested it with a every FPS from 0-300, it just maintains cyan above 60fps. 

The colors get clamped.